### DISCUSSION TITLE
A scheme that tries to protect prices

### DISCUSSION THREAD

**Discussion by goingsnginx** [sunflower-land/sunflower-land#1347] [Ideas]
At present, a large amount of funds of the project are sucked up by scripts that only farm and collect eggs. For such low-threshold accounts, a penalty mechanism should be added.
The performance of the low threshold script account: finish the seed, then withdraw and sell the tokens, maybe it will be synchronized once every 5 days, such a synchronization fee has almost no cost to them.
Features of low threshold script account: Participate in as few activities as possible, such as it will not participate in tree cutting and mining, its behavior is very single, maybe it may not participate in land expansion in the future, just rely on a piece of land to keep sucking blood

Principle: For accounts that do not participate in other activities for a long time, lower the threshold for synchronization, or increase the price of synchronization
plan 1:
There are different behavior points for game activities, such as tree cutting, mining, pasture, etc.
for example:
Farming: 1
Egg: 1
Tree chop: 2
Mining: 3

User quality = points / behavior frequency
By ranking users, those with lower rankings need to lower the threshold for synchronization, or increase the price of synchronization
Can add some grace period to newly created land

Scenario 2:
for land expansion
Users need to complete at least one land expansion every month, and if they exceed 20 (can be changed), they will no longer be punished forever.
New users are given a forbearance period for the first month. If the land expansion is not completed in the second month, the synchronization fee will increase by 20% to $0.12. If the land expansion has not been completed in the third month, the synchronization fee will continue to increase by 20%. It becomes 0.12*1.2=0.144 dollars. If the user completes the first land expansion in the third month, the synchronization fee will be reduced by 20% on the previous basis to 0.12 dollars
If the user completes 5 land expansions, they can not be penalized for 5 months
Real players will still complete land expansion even if they are slow

The purpose of this is to screen out unreal players through long-term single behavior as much as possible, and give certain penalties to increase their costs and affect their selling costs, thereby protecting the token price

**Comment by adamhannigan** [comment]
@goingsnginx Thanks for raising this - land expansion will incorporate some similar mechanics where buildings will break down + droughts will begin for people who appear to be draining the game of resources and not playing "normally" - i.e scripting behaviour

In total, the game will become a lot more balanced. You will need to engage in collect a range of resources to prosper in Sunflower Land. Our current version can be too easily single focussed to earn SFL.

**Reply by goingsnginx** [reply]
I know that your original intention was to make the poor as fair as possible, but unfortunately they were all exploited by scripted accounts, and I am concerned about the price of SFL when I know that some players are running hundreds of no ui accounts on one server.
But I really believe in the sunflower community, I believe you will solve all this

**Reply by goingsnginx** [reply]
I think the stop price of the script account is very important, it profoundly affects the selling pressure of the token, if they just take a part of the profit to maintain their own game state, it seems that it does not have much impact, as long as it is profitable , it can still continue to do so.
So I think it still makes sense to increase the forced consumption cost of accounts with a single behavior

**Reply by Yougottabekiddingme123** [reply]
> @goingsnginx Thanks for raising this - land expansion will incorporate some similar mechanics where buildings will break down + droughts will begin for people who appear to be draining the game of resources and not playing "normally" - i.e scripting behaviour
> 
> In total, the game will become a lot more balanced. You will need to engage in collect a range of resources to prosper in Sunflower Land. Our current version can be too easily single focussed to earn SFL.

Happy to hear this. Looking forward to the new mechanisms that protect the game.

**Reply by adamhannigan** [reply]
I 100% agree.

This is one of our main focusses at the moment with anti-botting/scripting and prevent these players from draining the economy.

**Reply by goingsnginx** [reply]
I consider the level of harm to the product to be prioritized:
1. No ui script account, can be automatically planted, 3-5 days to complete a synchronization, use a third party to pass Google verification, and automatically withdraw and sell. A $20 linux server can run hundreds of accounts, and the cost of getting sfl is very low, meaning unless sfl goes to 0, it's likely to run continuously
2. Script accounts for running browsers, this cost is slightly higher, maybe a $20 windows machine can run 10 accounts, which means their shutdown price is also slightly higher, each account invests in the server every month The cost is $2, and the agency cost is $1. If you synchronize once every four days, the monthly synchronization cost is $0.7, and they can get an average of 3sfl per day, which is 90sfl a month, and the cost of obtaining sfl is $0.041, which is low At 0.041 they will stop running, in fact maybe below 0.05 they will choose to stop running
3. Artificial players, but with multiple accounts. The cost of labor will always be a lot higher, although their cost of acquiring tokens is very low, but they think their labor cost will be slightly higher, so their price expectations will also be slightly higher, too low may not form a big deal of selling pressure.

my point of view:
1. Comprehensively crack down on no ui script accounts, and comprehensively clean up accounts that do not interact with ui
2. A lot of random interactive verification elements are added for users with long-term single behavior, which increases their development cost. This is only for users with low long-term behavior scores, but not for users with high behavior scores.
3. Increase penalties for users with the bottom 10% behavior points, such as increasing synchronization costs

**Reply by goingsnginx** [reply]
Adam, the price of sfl these days has fully proved that script accounts are far more harmful to the project than manual multi-account players. When they approached the script account shutdown price of 0.05, sfl stopped falling. When they restarted the script account, sfl began to fall.

Their harm is too great. At least during the period before the halving, any incremental capital entry will end at a loss, which means that sfl has no way to carry more funds. Even though sfl was pulled to $1 by a lot of force, they could still suck sfl down to $0.05 with a bunch of new script accounts.

In addition, I have a big question, although the initial land weight of the new version is reduced, they can still buy other people's land that has participated in land expansion at a low cost, just as now they are buying other people's old land, as long as Profitable, they can continue doing the same things they are doing now, and I still don't see if building durability and fertilizers are any more limiting for them.

Because many scripted accounts behave like real people, some accounts are very cautious and don't sell sfl, which means they're hard to spot. So I think it is very necessary to distinguish between artificial multi-account players and script accounts in the future. Only by killing script accounts as much as possible, sfl can carry more funds.

Suggest:
1. It is still the same as above to rate the behavior
2. Regularly add a new random behavior verification for the bottom 10% of users, a verification that has not appeared before, distinguish between manual multi-account accounts and script accounts, and kill script accounts
3. Increase their sync cost (just to increase shutdown price)
4. Bravely close those suspicious accounts, unless they go to discord to appeal, because the script account appeal cost will be very high

**Comment by johnnsie** [comment]
I wholeheartedly disagree with the anti botting mechanisms described in this thread. This would just results in endless back and forths of hard coded rules and changes ever more inneficient. 

Bots just automate a simple game style, which is how the game became very popular in the first place and is proof the game can be grinded with very low entry cost while remaining a profitable p2e. If you 'price out' bots, you'll effectively price out a good chunk of the player base.

The solution is to complexify the gameplay so that botting becomes way less profitable than smart decisions.

**Comment by goingsnginx** [comment]
I have some confusion about the new version of the economic system
First, let me explain what I initially thought were the four changes to the new system
1. The new version added a persistent land expansion loop and the bumpkin's loop
2. Reduce the weight of the initial land. Unless they spend a few dollars to open a few new land, the initial single -day output decreases by default. Since bots have fixed monthly expenses, the cost of producing SFL will increase, which is good for the price of sfl
3. The probability of  the green Amulet in new version was increased, which raised the cap on premium accounts and made multiple accounts less cost-effective in comparison. This does improve a problem in V2, where multiple accounts were more attractive than premium accounts.
4. The 14-day limit on the secondary market and the lack of limits on internal transfers are a clever restriction that raises the cost and risk for land traders and large numbers of bot accounts.

Until I found grub shop, a function that  has a long time cd and the average return is very high, so I have to confuse me

As we all know, the V2 initially entered a lot of funds, but because of the low threshold for acquiring sfl, the speed at which users earn sfl was far slower than the speed at which bot accounts were stealing market value, so even though many players kept buying on dips, they were still losing, until price reached 0.05, a calculable shutdown price for bot accounts. That's when sfl came back to life, and when it halved again, the price became 0.1, which is why we can see that even in a bear market, sfl are still steadily holding at around 0.1.As far as I know, at this time, the single day sfl output of the bot accounts don't exceed 1.5.This means that if you want to keep the price of sfl in a higher position, the single day output needs to be smaller

The high cost effective props such as pumpkin soup will be greater to the help of real users? The cost of pumpkin, which is only cost $ 1, can allow a bot account to reach a good level, and get many skills. For me, I often don't make pumpkin soup in order to save time. The cost of upgrading the bot account to the same level may be 1/3 to 1/2 of most real players

I think grub shop  brought some problems
The first is to make multiple ordinary accounts re -have advantages, which is equivalent to not doing any improvement. Although premium accounts can bring a lot of output, only a small part can be sold more to the grub shop. The cost performance is not as good as multiple ordinary accounts. For ordinary accounts, every ordinary account can sold to the grub shop to enjoy ultra -high returns.
The second is the low threshold,at the same time, a single day single account has a large number of outputs, based on the previous explanation the bot account produces a single daily output of less than 1.5sfl to maintain the price of 0.1. Without the entry of new users and new funds, put all existing accounts into the new version. The price of sfl will run below 0.05. This is undoubtedly a huge harm to those sfl holders. In my opinion, This will almost repeat the same mistakes. 
First of all, a large number of new players and money poured in, and then the money was taken away by a large number of bot accounts. Finally, the stop price of the new version of the bot account was touched, perhaps 0.04, or 0.03. The price of the game returns to a lower point. Although I know you want to please new users, it seems that the final result is to hurt users who really pay money. Although the expansion of the bot account and the upgrade requires some costs. Even if it is calculated at the current price, it only takes less than 5 US dollars to restore the production capacity of V2. However, there is a daily output that exceeds 4sfl, and the cost can be recovered in one week. Those who spend large costs buy nft and buy sfl are facing a high risk,longer cycles and losses possibilities.

Of course, it is necessary to admit that many vegetables used for upgrading also have good benefits in the formal version, perhaps double or twice, but it is far from 4 times that of the grub shop.And their profit sources are different. Vegetables sold in the secondary market are always less harmful.

I have seen some people with thousands of accounts, and I can predict that they will take a large amount of money from new players in the new version.
The function of low threshold and high cost performance is never really enjoyed by real users. The speed of bot accounts stealing market value is much faster than that of real users acquiring tokens
